### DISCUSSION TITLE
How to implement in PyQt aplication?

### DISCUSSION THREAD

**Discussion by dRAT3** [fief-dev/fief#169] [Q&A]
I need to know how to use the login in a pyqt app, can I just set the client to Public and hardcode the CLIENT_SECRET and CLIENT_ID in the qt app? Also I want to show only a login not that first screen with the id and secret.

**Comment by dRAT3** [comment]
Also do I need to self-host to access roles, I can't click on the lock icon..

**Reply by frankie567** [reply]
Regarding this, you probably hit issue #54. Try to enlarge your window to see the full menu.

**Comment by fief-bailiff** [comment]
Hail, @dRAT3 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
If you build a desktop app, you should create a "Public" app, where the client secret is not needed (we can't assume it's safe since you ship the code to the user).

In this case, PKCE is required for extra security: https://docs.fief.dev/going-further/pkce/

The Python SDK has functions to help you work with PKCE: https://fief-dev.github.io/fief-python/fief_client/pkce.html

If you need inspiration, you can probably have a look at our Python CLI integration: https://github.com/fief-dev/fief-python/blob/main/fief_client/integrations/cli.py
